bioinf_mega_tools.py

```diff
@@ -1,47 +1,84 @@
-from scripts.filter_params import is_filter_gc
-from scripts.filter_params import is_filter_length, is_filter_quality
-import scripts.dna_rna_tools
-import scripts.rw_fastq
+from abc import ABC
+from Bio import SeqIO
+from Bio.SeqUtils import GC
+import os, os.path
 
-operations = {
-    "transcribe": scripts.dna_rna_tools.transcribe,
-    "reverse": scripts.dna_rna_tools.reverse,
-    "complement": scripts.dna_rna_tools.complement,
-    "reverse_complement": scripts.dna_rna_tools.reverse_complement,
-}
 
+class BiologicalSequence(ABC):
+    def __init__(self, sequence: str):
+        self.sequence = sequence.upper()
 
-def run_dna_rna_tools(*args: str) -> str | dict:
-    """
-    The function returns DNA or RNA sequence
-    after one of the following operations:
-    - transcribe (works only for DNA)
-    - reverse
-    - complement
-    - reverse_complement (reverse and complement together)
+    def __len__(self):
+        return len(self.sequence)
 
-    Args:
-    last one - operation name
-    others - DNA/RNA sequence or sequences to operate
-    (if the value is not DNA/RNA returns a warning, not a sequence)
-    """
-    seqs = args[:-1]
-    operation = args[-1]
-    results = []
+    def __getitem__(self, index):
+        return self.sequence[index]
+
+    def __str__(self):
+        return self.sequence
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}: {self.sequence}"
+
+    @staticmethod
+    def check_alphabet(sequence, alphabet):
+        return set(sequence).issubset(alphabet)
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    complement_nucl = {"A": "T", "T": "A", "G": "C", "C": "G", "U": "A"}
+
+    def complement(self):
+        return self.__class__("".join([self.complement_nucl[i] for i in self.sequence]))
 
-    for i in range(len(seqs)):
-        is_dna = scripts.dna_rna_tools.is_dna
-        is_rna = scripts.dna_rna_tools.is_rna
-        if not is_dna(seqs[i]) and not is_rna(seqs[i]):
-            seqs[i] = "Warning, input is not DNA/RNA"
+    def reverse(self):
+        return self.__class__(self.sequence[::-1])
 
-    for seq in seqs:
-        result = operations[operation](seq)
-        results.append(result)
+    def reverse_complement(self):
+        return self.__class__(self.complement().reverse().sequence)
```

Comment:
У многих наоборот, возвращает строку, а тут overkill - и reverse, и complement и так возвращают объекты своих классов, доставать из них строку, чтобы опять сделать объект класса смысла нет.
```suggestion
        return self.complement().reverse()
```